common/util.go

```diff
@@ -668,3 +669,54 @@ func GetPayloadsMapSize(data map[string]*commonpb.Payloads) int {
 
 	return size
 }
+
+// GetWorkflowExecutionTimeout gets the default allowed execution timeout or truncates the requested value to the maximum allowed timeout
+func GetWorkflowExecutionTimeout(
+	namespace string,
+	requestedTimeout *time.Duration,
+	getDefaultTimeoutFunc dynamicconfig.DurationPropertyFnWithNamespaceFilter,
+	getMaxAllowedTimeoutFunc dynamicconfig.DurationPropertyFnWithNamespaceFilter) *time.Duration {
```

Comment:
```suggestion
	getMaxAllowedTimeoutFunc dynamicconfig.DurationPropertyFnWithNamespaceFilter) time.Duration {
```
I would return struct here and get rid of dereferencing on caller side. The reason we have all those `*time.Time` is because protoc compiles it this way. I would use pointers in our code only if it makes logical sense (like time or duration can be `nil`). Everywhere else I would use just structs.

Reply:
good catch